### DISCUSSION TITLE
Allow to Peform Initial Reduction and Input Application of the Contract

### DISCUSSION THREAD

**Discussion by paluh** [marlowe-lang/marlowe-cardano#640] [Ideas]
Ability to reduce the contract and also perform inputs can be really useful:

* we can save resources on the chain
* we could do the contract settlement together with some initial deposits or funding

As a part of this proposal we have to also expose `initialState` on the REST API level (which should be part of if anyway).

**Comment by bwbush** [comment]
Yes. It slightly increases the risk of the later parts of the contract exceeding the Plutus execution budget, so we need to make sure that the safety checks rigorously forbid dangerous initial states.
